src/PoolManager.sol

```diff
@@ -17,13 +17,13 @@ import {IHookFeeManager} from "./interfaces/IHookFeeManager.sol";
 import {IPoolManager} from "./interfaces/IPoolManager.sol";
 import {ILockCallback} from "./interfaces/callback/ILockCallback.sol";
 import {Fees} from "./Fees.sol";
-import {Claims} from "./Claims.sol";
+import {ERC6909} from "solmate/tokens/ERC6909.sol";
```

Comment:
I'd rather we copy and paste this contract in, and rewrite the `_mint` and `_burn` functions to not use `totalSupply`, end remove the `totalSupply` mapping entirely. It feels wrong that we still have that mapping but its always 0. And I think it'll be neater if the PoolManager can just call `_mint` and _`burn` instead of having the logic locally

Reply:
oh i see @jtriley-eth is doing a PR to do this within solmate itself... i guess we will hold this PR til thats done

Reply:
Kind of agree. So either we hold this PR until its completed in solmate or copy/paste/credit but modify totalSupply and then later use the solmate package when its ready. Kind of in favor of the latter so we can unblock other PRs for core.

Reply:
Sure I can copy and paste the solmate impl into here

Reply:
Using solmate impl